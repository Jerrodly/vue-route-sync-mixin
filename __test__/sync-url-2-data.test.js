/**
 * Summary.
 *  main test file.
 *
 * Description
 *  This is the main test file
 *
 * @file   This main test file for class vue-route-sync-mixin.
 * @author teeeemoji.
 * @since 2020-1-5
 */
import {createLocalVue, mount} from '@vue/test-utils'
import VueRouter from 'vue-router'
import Comp from './test-material/component.vue'
import App from './test-material/app.vue'
import EmptyComp from './test-material/empty.vue'

describe('initial data from url', function () {
  jest.setTimeout(500)
  let wrapper = null
  let comp = null

  beforeAll(() => {
    const localVue = createLocalVue()
    const routes = [{path: '/', component: Comp}]
    const router = new VueRouter({
      routes
    })
    localVue.use(VueRouter)
    wrapper = mount(App, {
      localVue,
      router,
      attachToDocument: true
    })
    wrapper.vm.$router.push({
      query: {
        ps: '1__1000',
        pn: '1__2000',
        n: '2__ztcccc',
        c: '3__%7B"start"%3A"1223456","end"%3A"6543221"%7D',
        d: '3__%5B10,20,30,"four"%5D',
        u: '1__100',
        s: '5__true',
        a: '123123'
      }
    })
    comp = wrapper.vm.$refs.comp
  })

  afterAll(() => {
    wrapper.destroy()
    wrapper = null
  })

  it('initial undefined data', function () {
    expect(comp.pagination.pageNum).toMatchSnapshot()
    expect(comp.pagination.pageSize).toMatchSnapshot()
    expect(comp.name).toMatchSnapshot()
    expect(comp.smoke).toMatchSnapshot()
    expect(comp.departments).toMatchSnapshot()
    expect(comp.createDate).toMatchSnapshot()
  })
})
