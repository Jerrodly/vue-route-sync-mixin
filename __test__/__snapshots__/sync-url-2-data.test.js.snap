// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial data from url initial undefined data 1`] = `2000`;

exports[`initial data from url initial undefined data 2`] = `1000`;

exports[`initial data from url initial undefined data 3`] = `"ztcccc"`;

exports[`initial data from url initial undefined data 4`] = `true`;

exports[`initial data from url initial undefined data 5`] = `
Array [
  10,
  20,
  30,
  "four",
]
`;

exports[`initial data from url initial undefined data 6`] = `
Object {
  "end": "6543221",
  "start": "1223456",
}
`;
